packages/types/src/extrinsic/Extrinsic.ts

```diff
@@ -36,6 +37,8 @@ const VERSIONS = [
 export { EXTRINSIC_VERSION as LATEST_EXTRINSIC_VERSION } from './v4/Extrinsic';
 
 abstract class ExtrinsicBase<A extends AnyTuple> extends Base<ExtrinsicVx | ExtrinsicUnknown> {
+  protected hashCache?: CodecHash;
```

Comment:
Not 100% sure why this is protectected? Doesn't the `#hashCache` make more sense?

If `protected` is needed, just do it as `_hashCache`, just to indicate it is internal. (minor)

Reply:
This is `ExtrinsicBase`, it doesn't have signature-related methods on it, they are in `GenericExtrinsic`, which extends `ExtrinsicBase`, so I had to make property protected such that code in `GenericExtrinsic` can access it.